drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1756,13 +1725,20 @@ int usb_dc_ep_write(const uint8_t ep, const uint8_t *const data,
 	}
 
 	ep_ctx->write_in_progress = true;
-	NRFX_USBD_TRANSFER_IN(transfer, ep_ctx->buf.data, ep_ctx->buf.len, 0);
+	NRFX_USBD_TRANSFER_IN(transfer, data, data_len, 0);
```

Comment:
I suspect there are buffer alignment requirements, should it be checked?

Reply:
Im not sure, I'll check that.

Reply:
I can't find any requirements for buffer alignment. Do You have something in mind?

Reply:
I've briefly checked it with hid-mouse sample where the data are transfered directly using `usb_dc_ep_write` function. It works well. The data I was sending were not aligned 4.